l10n_ch_payment_register/models/payment_register.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (c) 2015 brain-tec AG (http://www.braintec-group.com)
+#    All Right Reserved
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import time
+from openerp import models, fields, api
+
+
+class PaymentRegister(models.Model):
+    _name = 'payment.register'
+    _description = 'Payment Register'
+    _rec_name = 'reference'
+    _order = 'id desc'
+
+    @api.multi
+    @api.depends('line_ids.amount')
+    def _compute_total(self):
+        for payment_register in self:
+            total = 0
+            for line in payment_register.line_ids:
+                total += line.amount
```

Comment:
here you could use the `mapped` recordset method

    sum(payment_register.mapped('line_ids.amount'))